lux/action/temporal.py

```diff
@@ -0,0 +1,133 @@
+
+    Returns
+    -------
+    recommendations : Dict[str,obj]
+            Object with a collection of visualizations that result from the Temporal action.
+    """
+    visuals = []
+    recommendation = {
+        "action": "Temporal",
+        "description": "Show trends over <p class='highlight-descriptor'>time-related</p> attributes.",
+        "long_description": "Temporal displays line charts for all attributes related to datetimes in the dataframe.",
+    }
+
+    for c in ldf.columns:
+        if ldf.data_type[c] == "temporal":
+            try:
+                generated_vis = create_temporal_vis(ldf, c)
+                visuals.extend(generated_vis)
+            except (ValueError, AttributeError, KeyError):
+                pass
+
+    # If no temporal visualizations were generated via parsing datetime, fallback to default behavior.
+    if len(visuals) == 0:
+        intent = [lux.Clause("?", data_type="temporal")]
+        intent.extend(utils.get_filter_specs(ldf._intent))
+        vlist = VisList(intent, ldf)
+        for vis in vlist:
+            vis.score = interestingness(vis, ldf)
+    else:
+        vlist = VisList(visuals)
+        recommendation["long_description"] += (
+            " The scoring will display all overall visualizations first,"
+            + "all the (year) visualizations second, all the (month) visualizations third, and finally all the (day) visualizations."
+        )
+
+    # Doesn't make sense to generate a line chart if there is less than 3 datapoints (pre-aggregated)
+    if len(ldf) < 3:
+        recommendation["collection"] = []
+        return recommendation
+    vlist.sort()
+    recommendation["collection"] = vlist
+    return recommendation
+
+
+def create_temporal_vis(ldf, col):
+    """
+    Creates and populates Vis objects for the different time-scale visualizations of the provided temporal column.
+
+    Parameters
+    ----------
+    ldf : lux.core.frame
+            LuxDataFrame with underspecified intent.
+
+    col : str
+            Name of temporal column.
+
+    Returns
+    -------
+    visuals : [Vis]
+            Collection of Vis objects.
+    """
+    formatted_date = pd.to_datetime(ldf[col], format="%Y-%m-%d")
+
+    overall_vis = Vis([lux.Clause(col, data_type="temporal")], source=ldf, score=4)
+
+    year_col = col + " (year)"
+    year_df = LuxDataFrame({year_col: pd.to_datetime(formatted_date.dt.year, format="%Y")})
+    year_vis = Vis([lux.Clause(year_col, data_type="temporal")], source=year_df, score=3)
+
+    month_col = col + " (month)"
+    month_df = LuxDataFrame({month_col: pd.to_datetime(formatted_date.dt.month, format="%m")})
+    month_vis = Vis([lux.Clause(month_col, data_type="temporal")], source=month_df, score=2)
+
+    day_col = col + " (day)"
+    day_df = LuxDataFrame({day_col: pd.to_datetime(formatted_date.dt.day, format="%d")})
+    day_vis = Vis([lux.Clause(day_col, data_type="temporal")], source=day_df, score=1)
```

Comment:
I find it odd that we're building the data frame and vis directly from the action class rather than using classes like `VisList` and `PandasExecutor`. At the same time, I see where you were coming from (we're essentially extending making copies of the same data frame just using different time scales). I'll leave my comment here in case Doris has any input on that!

Reply:
Hi @micahtyong, thanks for the review! I'm not sure why the y-axis seems to have a minimum far below the data points. The fix in this PR relies on the existing functionality to populate the visualizations; it appears that the same y-axis range is being produced even when the changes from this PR are not added -- for instance, the flights dataset -- but I can take a closer look at it!